File: module-code/app/securesocial/core/DefaultResolver.scala
package securesocial.core
import securesocial.core.java.ResolverHandler
import play.api.mvc.Call
import play.api.mvc.Request

abstract class DefaultResolver(loginCall:Call, logoutCall:Call) extends ResolverHandler.Resolver {

	override def getLoginUrl():String = {
		loginCall.url
	}

	override def getLoginUrlAbsolute(req: Request[_]):String = {
		loginCall.absoluteURL(ResolverHandler.req2req(req))
	}

	override def getLogoutUrl():String = {
		logoutCall.url
	}

	override def getLogoutUrlAbsolute(req: Request[_]):String = {
		logoutCall.absoluteURL(ResolverHandler.req2req(req))
	}
	
	override def getAuthenticateUrl(provider: String):String = {
		throw new RuntimeException("You must implement getAuthenticateUrl(String)")
	}

	override def getAuthenticateUrlAbsolute(provider: String,
			req: Request[_]):String = {
		throw new RuntimeException("You must implement getAuthenticateUrlAbsolute(String, play.api.mvc.Request)");
	}
}

File: module-code/app/securesocial/core/java/DefaultResolver.java
package securesocial.core.java;

import securesocial.core.java.ResolverHandler.Resolver;
import play.mvc.Call;
import play.mvc.Http.Request;

public abstract class DefaultResolver extends Resolver {

	private final play.mvc.Call loginCall;
	private final play.mvc.Call logoutCall;
	
	public DefaultResolver(final Call login, final Call logout) {
		loginCall = login;
		logoutCall = logout;
	}
	@Override
	public String getLoginUrl() {
		return loginCall.url();
	}

	@Override
	public String getLoginUrlAbsolute(final Request req) {
		return loginCall.absoluteURL(req);
	}

	@Override
	public String getLogoutUrl() {
		return logoutCall.url();
	}

	@Override
	public String getLogoutUrlAbsolute(final Request req) {
		return logoutCall.absoluteURL(req);
	}
	
	@Override
	public String getAuthenticateUrl(final String provider) {
		throw new RuntimeException("You must implement getAuthenticateUrl(String)");
	}

	@Override
	public String getAuthenticateUrlAbsolute(final String provider,
			final Request req) {
		throw new RuntimeException("You must implement getAuthenticateUrlAbsolute(String, play.mvc.Http.Request)");
	}

}
